src/utils/convert.ts

```diff
@@ -1377,6 +1461,380 @@ async function processM2NRehypeImageNodes(
+      hx = getTextFromNode(node);
+    } else {
+      hx = getTextFromNode(node);
+    }
+
+    node.type = "text";
+    node.value = hx;
+  }
+}
+
+async function processN2LRehypeLinkNodes(nodes: string | any[]) {
+  if (!nodes.length) {
+    return;
+  }
+  for (const node of nodes) {
+    node.type = "text";
+    node.value = `\\href{${node.properties.href}}{${getTextFromNode(node)}}`;
+  }
+}
+
+async function processN2LRehypeListNodes(nodes: string | any[]) {
+  if (!nodes.length) {
+    return;
+  }
+  for (const node of nodes) {
+    const item_str: string[] = [];
+    for (const itemKey in node.children) {
+      const itemNode = node.children[itemKey];
+      if (itemNode.type === "element" && itemNode.tagName === "li") {
+        item_str.push(getTextFromNode(itemNode));
+      }
+    }
+    // Zotero.debug(item_str);
+    const join_str = item_str.join("\n\\item");
+    let listTemplate;
+    if (node.tagName === "ul") {
+      const ulTemplate = `\\begin{itemize}\n\\item ${join_str} \n\\end{itemize}`;
+      listTemplate = ulTemplate;
+    } else if (node.tagName === "ol") {
+      const olTemplate = `\\begin{enumerate}\n\\item ${join_str} \n\\end{enumerate}`;
+      listTemplate = olTemplate;
+    }
+
+    node.type = "text";
+    node.value = listTemplate;
+  }
+}
+
+async function processN2LRehypeTableNodes(nodes: any[]) {
+  if (!nodes.length) {
+    return;
+  }
+  for (const node of nodes) {
+    if (node.type === "element" && node.tagName === "table") {
+      const latexTable = convertHtmlTableToLatex(node.children[1]);
+      node.type = "text";
+      node.value = latexTable;
+    }
+  }
+}
+
+function convertHtmlTableToLatex(tableNode: any): string {
+  let colCount = 0;
+  let rowCount = 0;
+  const tableData: string[][] = [];
+
+  for (const child of tableNode.children) {
+    if (child.type === "element" && child.tagName === "tr") {
+      rowCount++;
+      const rowData: string[] = [];
+      for (const td of child.children) {
+        if (
+          td.type === "element" &&
+          (td.tagName === "td" || td.tagName === "th")
+        ) {
+          colCount = Math.max(colCount, rowData.length + 1);
+          rowData.push(getTextFromNode(td));
+        }
+      }
+      tableData.push(rowData);
+    }
+  }
+
+  let columnFormat = "|";
+  for (let i = 0; i < colCount; i++) {
+    columnFormat += "l|"; // Assuming left alignment for all columns
+  }
+
+  const latexRows: string[] = [];
+  latexRows.push("\\hline");
+  for (const row of tableData) {
+    const rowContent = row
+      .map((cell) => cell.replace(/_/g, "\\_").replace(/&/g, "\\&"))
+      .join(" & ");
+    latexRows.push(rowContent + " \\\\");
+    latexRows.push("\\hline");
+  }
+
+  const latexTableTemplate = `\\begin{table}[htbp]\n\\centering\n\\caption{Caption}\n\\label{tab:simple_table}\n\\begin{tabular}{${columnFormat}}
+${latexRows.join("\n")}
+\\end{tabular}\n\\end{table}`;
+
+  return latexTableTemplate;
```

Comment:
The variable naming is confusing here. Since it's a return value, it's the actual string we want, not a template.